src/cpp/multisolver_interface/SolverAbstract.cpp

```diff
@@ -0,0 +1,141 @@
+#include "antares-xpansion/multisolver_interface/SolverAbstract.h"
+
+namespace
+{
+bool verifyProblemDimensions(const SolverAbstract* merged, const SolverAbstract* master)
```

Comment:
This function is general to compare any problem, the names `merged` and `master` are misleading, I would use more generic ones, such as `problem1` and `problem2` (or any better suggestion, `lhs` and `rhs` ? ;)). Also I would name the function `areProblemsDimensionEqual` because out of the context of the test, there is no reason that `verify` aims at dimensions equality in a generic context. This remark applies for all subsequent functions.

Reply:
Nice suggestion